File: Guanabara - Java/Meus exercícios POO/Aula02/src/main/java/aula02/Caneta.java
package aula02;
public class Caneta {
    String modelo;
    String cor;
    float ponta;    //5 atributos
    int carga;
    boolean tampada;
    
    void status() {
        System.out.println("Uma caneta " + this.cor);
        System.out.println("Está tampada? " + this.tampada);
        System.out.println("Modelo " + this.modelo);
        System.out.println("Ponta " + this.ponta + "mm");
        System.out.println("Carga " + this.carga + "%");
    }
    void rabiscar() {
        if (this.tampada == true) {
            System.out.println("[ERRO! Não é possivel rabiscar pois a caneta está tampada]");
        } else {
            System.out.println("Estou rabiscando");
        }
    }
    void tampar() {
        this.tampada = true;
        System.out.println("Tampada? " + true);
    }
    void destampar() {
        this.tampada = false;
        System.out.println("Destampada? " + true);
    }
}

File: Guanabara - PHP/exercicios-PHPOO/Aula03/Caneta.php
<?php
class Caneta {
    public $modelo;    //adicionando atributos/componentes na classe
    public $cor;
    private $ponta; //atributo privado: só posso mexer se estiver dentro dessa classe
    private $carga; 
    protected $tampada; //atributo protegido: só posso editar nessa classe ou em suas filhas.         dúvida:(((ou tbm através de funções???)))
    
    function rabiscar() {
    if ($this-> tampada == true) { //se a caneta estiver tampada
        echo "<p>ERRO! A caneta está tampada, não posso rabiscar</p>";
    } else {
        echo "<p>Estou rabiscando...</p>";
        }        
    }
    public function tampar () { //não declarar a visibilidade ainda mantém o método como publico (default)
        $this->tampada = true;
    }
    public function destampar () {
        $this->tampada = false;
    }
}
